vizro-core/src/vizro/actions/_actions_utils.py

```diff
@@ -66,34 +63,97 @@ def _apply_filters(
     return data_frame
 
 
-def _apply_filter_interaction(
-    data_frame: pd.DataFrame,
-    ctds_filter_interaction: List[CallbackTriggerDict],
-    target: str,
+def _apply_chart_filter_interaction(
+    data_frame: pd.DataFrame, target: str, ctd_filter_interaction: Dict[str, CallbackTriggerDict]
```

Comment:
Curious, what made the typing change here? This originates from `ctx` which is not in our code-base right? So why has this changed its type?

Reply:
It's caused by changes made in the `_callback_mapping_utils.py  _get_inputs_of_figure_interactions function`. Its `return value` is changed since some actions required multiple component properties to be propagated. So, I "packed" them into Dict[str, CallbackTriggerDict] format.

Reply:
Yes exactly. It has the same format for all of currently predefined actions. Dictionary with keys ("filters", "parameters", "filter_interaction", "theme_selector") 

You can find more on  [Dash flexible callback signatures](https://dash.plotly.com/flexible-callback-signatures#dictionary-grouping)